src/Microsoft.Health.Dicom.Core.UnitTests/Features/Retrieve/RetrieveMetadataServiceTests.cs

```diff
@@ -137,6 +137,7 @@ public async Task GivenRetrieveInstanceMetadataRequestForStudy_WhenIsSuccessful_
         RetrieveMetadataResponse response = await _retrieveMetadataService.RetrieveStudyInstanceMetadataAsync(_studyInstanceUid, ifNoneMatch, DefaultCancellationToken);
 
         Assert.Equal(response.ResponseMetadata.Count(), versionedInstanceIdentifiers.Count);
+        Assert.Equal(response.ResponseMetadata.Count(), _dicomRequestContextAccessor.RequestContext.PartCount);
```

Comment:
nit: I think this equality check is backwards (at least for the purpose of the possible error message if it fails). I think it should be `Assert.Fails(<expected>, <actual>)`

Reply:
I want to assert that the PartCount has been updated to the response count - so in this case wouldn't the "actual" be PartCount and the "expected" be response count?

Reply:
Ah yes, I see. You're correct